app/src/main/java/org/wikipedia/WikipediaApp.kt

```diff
@@ -251,16 +251,20 @@ class WikipediaApp : Application() {
     }
 
     fun resetAfterLogOut() {
-        AccountUtil.removeAccount()
-        Prefs.isPushNotificationTokenSubscribed = false
-        Prefs.pushNotificationTokenOld = ""
-        Prefs.tempAccountWelcomeShown = false
-        Prefs.tempAccountCreateDay = 0L
-        Prefs.tempAccountDialogShown = false
-        SharedPreferenceCookieManager.instance.clearAllCookies()
-        AppDatabase.instance.notificationDao().deleteAll()
-        FlowEventBus.post(LoggedOutEvent())
-        L.d("Logout complete.")
+        MainScope().launch(CoroutineExceptionHandler { _, t ->
+            L.e(t)
+        }) {
+            AccountUtil.removeAccount()
+            Prefs.isPushNotificationTokenSubscribed = false
+            Prefs.pushNotificationTokenOld = ""
+            Prefs.tempAccountWelcomeShown = false
+            Prefs.tempAccountCreateDay = 0L
+            Prefs.tempAccountDialogShown = false
+            SharedPreferenceCookieManager.instance.clearAllCookies()
+            AppDatabase.instance.notificationDao().deleteAll()
```

Comment:
^ It looks like this is the only Dao call that is made in this function, and it doesn't really impact the Csrf logic. So, we can just run this single line as a miniature coroutine, and leave the rest of this function as before.

Reply:
Sure! Done.